src/osd/ECCommon.cc

```diff
@@ -524,6 +566,46 @@ struct ClientReadCompleter : ECCommon::ReadCompleter {
   ECCommon::ClientAsyncReadStatus *status;
 };
 
+bool ECCommon::ReadPipeline::should_partial_read(
+  const hobject_t &hoid,
+  std::list<ECCommon::ec_align_t> to_read,
+  const std::set<int> &want,
+  bool fast_read,
+  bool for_recovery)
+{
+    // Don't partial read if we are doing a fast_read
+    if (fast_read) {
+      return false;
+    }
+    // Don't partial read if the EC isn't systematic
+    if (!ec_impl->is_systematic()) {
+      return false;
+    }
+    // Don't partial read if we have multiple stripes
+    if (to_read.size() != 1) {
+      return false;
+    }
```

Comment:
I commented on this on the original PR -- I don't want there to be a distinction between a "partial read" and a read where all of the data chunks happen to be present.  The only real distinction should be that if the all-data-chunk read pattern isn't possible because of a degraded object, we fall back to reading full stripes in order to decode.

Distinctions mean branches, branches mean bugs.  If the read contains several full stripes but has a partial stripe on either side and the data chunks aren't degraded, I'd expect it to do partial reads at the wings with full reads in the middle.  Furthermore, I'd expect the actual logic to be pretty much identical to the case where the read boundaries happened to be stripe aligned.

Reply:
Makes sense.

Reply:
Eliminated the conditionality.